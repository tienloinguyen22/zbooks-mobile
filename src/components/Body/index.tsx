import { Body } from 'native-base';

export { Body };
