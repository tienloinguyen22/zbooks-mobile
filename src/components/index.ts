export * from './Body';
export * from './Button';
export * from './Card';
export * from './CardItem';
export * from './Container';
export * from './Content';
export * from './DatePicker';
export * from './ErrorBoundary';
export * from './ErrorText';
export * from './Field';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Image';
export * from './Input';
export * from './Item';
export * from './Label';
export * from './Left';
export * from './List';
export * from './ListItem';
export * from './Loading';
export * from './Picker';
export * from './Right';
export * from './ScrollView';
export * from './Text';
export * from './Touchable';
export * from './Thumbnail';
export * from './Tooltip';
export * from './View';
export * from './WithStore';
export * from './WithScreen';
export * from './Alert';
export * from './Animation';
export * from './AlertContainer';
