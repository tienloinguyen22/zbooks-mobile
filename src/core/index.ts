export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
