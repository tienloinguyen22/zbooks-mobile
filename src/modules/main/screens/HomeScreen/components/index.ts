export * from './NavigationSample';
export * from './IconSample';
export * from './RematchSample';
export * from './NotificationSample';
export * from './CrashSample';
export * from './PickerSample';
export * from './AnalyticsSample';
export * from './TextStyleSample';
