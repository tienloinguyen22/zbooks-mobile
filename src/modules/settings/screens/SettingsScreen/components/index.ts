export * from './SettingMenuItem';
export * from './LogoutModal';
