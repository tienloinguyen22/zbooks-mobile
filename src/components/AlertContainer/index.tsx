import React from 'react';
import { ViewStyle } from 'react-native';
import Modal from 'react-native-modal';
import { combineStyles, colors, THEME_DARK } from '@app/core';
import { useTranslation } from 'react-i18next';
import { useTheme } from '@app/hooks';
import { Icon } from '../Icon';
import { Button } from '../Button';
import { View } from '../View';
import { styles } from './styles';
import { Text } from '../Text';

export type AlertType = 'SUCCESS' | 'WARNING' | 'ERROR' | 'INFO' | 'DEFAULT';

export interface AlertAction {
  title: string;
  onPress: () => void;
  outline?: boolean;
}

interface AlertProps {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose: () => void;
  closeable?: boolean;
}

export const AlertContainer = (props: AlertProps): JSX.Element => {
  const { primaryColor, theme } = useTheme();
  const { t } = useTranslation();

  const backgroundAlertStyle = {
    backgroundColor: theme === THEME_DARK ? colors.lightBlack : colors.white,
  };

  const messageTextStyle = {
    color: theme === THEME_DARK ? colors.white : colors.black,
  };

  const closeModal = (): void => {
    props.closeable !== false && props.onPressClose();
  };

  let titleContainerStyle = combineStyles<ViewStyle>(styles.titleDefault, {
    backgroundColor: primaryColor,
  });
  let title = t('dialog.info');
  let iconName = 'information-outline';

  switch (props.type) {
    case 'WARNING':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.orange,
      });
      title = t('dialog.warning');
      iconName = 'alert-circle-outline';
      break;
    case 'ERROR':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.red,
      });
      title = t('dialog.error');
      iconName = 'alert-outline';
      break;
    case 'SUCCESS':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.green,
      });
      title = t('dialog.success');
      iconName = 'check';
      break;
    default:
      break;
  }

  props.title && ({ title } = props);

  return (
    <View style={styles.container}>
      <Modal isVisible={true} onBackButtonPress={closeModal} onBackdropPress={closeModal}>
        <View style={[styles.alertContainer, backgroundAlertStyle]}>
          <View style={titleContainerStyle}>
            <Icon name={iconName} color={colors.white} size={25} />
            <Text style={styles.titleText}>{title}</Text>
            <View style={styles.dialogIcon} />
          </View>
          <Text style={[styles.messageText, messageTextStyle]}>{props.message}</Text>
          {props.actions && (
            <View style={styles.buttonContainer}>
              {props.actions.map((item) => {
                return (
                  <Button outline={item.outline} onPress={item.onPress} key={item.title} style={styles.buttonDialog}>
                    <Text white={!item.outline} primary={item.outline}>
                      {item.title}
                    </Text>
                  </Button>
                );
              })}
            </View>
          )}
          {!props.actions && (
            <View style={styles.reserveContainer}>
              <Button onPress={props.onPressClose} style={styles.buttonDialog}>
                <Text white>{t('dialog.close')}</Text>
              </Button>
            </View>
          )}
        </View>
      </Modal>
    </View>
  );
};
