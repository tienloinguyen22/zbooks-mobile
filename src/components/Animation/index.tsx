import React from 'react';
import { Resource } from '@app/core';
import LottieView from 'lottie-react-native';

interface Props {
  width: number;
  height: number;
  source: Resource;
  autoPlay?: boolean;
  loop?: boolean;
}

export const Animation = (props: Props): JSX.Element => {
  return (
    <LottieView
      source={props.source}
      style={{
        width: props.width,
        height: props.height,
      }}
      autoPlay={props.autoPlay === true}
      loop={props.loop === true}
    />
  );
};
