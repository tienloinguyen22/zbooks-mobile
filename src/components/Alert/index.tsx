import React from 'react';
import RootSiblings from 'react-native-root-siblings';
import { AlertContainer, AlertAction, AlertType } from '../AlertContainer';

interface ShowParams {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose?: () => void;
  closeable?: boolean;
}

let alert: RootSiblings | undefined;
const hide = (): void => {
  alert && alert.destroy();
};

const show = (props: ShowParams): RootSiblings => {
  alert && hide();
  alert = new RootSiblings(<AlertContainer {...props} onPressClose={props.onPressClose || hide} />);
  return alert;
};

export const Alert = {
  show,
  hide,
};
