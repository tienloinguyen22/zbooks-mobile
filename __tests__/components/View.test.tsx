import React from 'react';
import { render } from '@testing-library/react-native';
import { View, Text, Icon } from '@app/components';
import { colors, THEME_DARK } from '@app/core';
import { mockTheme } from '../test_helpers';

beforeAll(() => {});

describe('components/View', () => {
  beforeEach(() => {
    mockTheme(THEME_DARK);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('renders successfully', async () => {
    const { baseElement } = render(<View />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with flex successfully', async () => {
    const viewText = 'view';
    const { baseElement } = render(
      <View flex={1}>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with center props successfully', async () => {
    const viewText = 'view';
    const { baseElement } = render(
      <View center>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('renders with center vertical props successfully', async () => {
    const viewText = 'view';
    const { baseElement } = render(
      <View centerVertical>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('renders with row props successfully', async () => {
    const viewText = 'view';
    const { baseElement } = render(
      <View row>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('renders with column props successfully', async () => {
    const viewText = 'view';
    const { baseElement } = render(
      <View column>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('renders with rowReverse props successfully', async () => {
    const viewText = 'view';
    const { baseElement } = render(
      <View rowReverse>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('renders with columnReverse props successfully', async () => {
    const viewText = 'view';
    const { baseElement } = render(
      <View columnReverse>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });
  it('renders with text children successfully', async () => {
    const viewText = 'view';
    const { baseElement, getByText } = render(
      <View>
        <Text>{viewText}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
    expect(getByText(viewText)).toBeDefined();
  });
  it('renders two text children successfully', async () => {
    const text1 = 'text1';
    const text2 = 'text2';
    const { baseElement, getByText } = render(
      <View>
        <Text>{text1}</Text>
        <Text>{text2}</Text>
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
    expect(getByText(text1)).toBeDefined();
    expect(getByText(text2)).toBeDefined();
  });

  it('renders with text and icon children successfully', async () => {
    const text1 = 'text1';
    const text2 = 'text2';
    const { baseElement } = render(
      <View>
        <Text>{text1}</Text>
        <Text>{text2}</Text>
        <Icon name='face-profile' color={colors.blue} />
      </View>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
