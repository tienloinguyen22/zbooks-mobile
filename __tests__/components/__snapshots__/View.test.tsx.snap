// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/View renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    />
  </View>
</View>
`;

exports[`components/View renders two text children successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        text1
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        text2
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with center props successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      center={true}
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with center vertical props successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      centerVertical={true}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with column props successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      column={true}
      style={
        Object {
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with columnReverse props successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      columnReverse={true}
      style={
        Object {
          "flexDirection": "column-reverse",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with flex successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      flex={1}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with row props successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      row={true}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with rowReverse props successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      rowReverse={true}
      style={
        Object {
          "flexDirection": "row-reverse",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with text and icon children successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        text1
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        text2
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#1A97FC",
              "fontSize": 25,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï™„
      </Text>
    </View>
  </View>
</View>
`;

exports[`components/View renders with text children successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        style={
          Object {
            "color": "white",
            "textTransform": "none",
          }
        }
      >
        view
      </Text>
    </View>
  </View>
</View>
`;
